### ISSUE TITLE
Account recovery: Add feature fields for web-app-based recovery tool

### ISSUE THREAD

**Issue by polymutex** [fluidkey/walletbeat#765]
<div type='discussions-op-text'>

<sup>Originally posted by **ren2140eth** May 20, 2026</sup>
## Motivation

While modeling Coinbase Smart Wallet's "recovery key" feature for Base App ([PR #740](https://github.com/walletbeat/walletbeat/pull/740)), we hit a gap in the current `GuardianPolicyType` enum. @polymutex's review pointed out that:

- The feature isn't `SECRET_SPLIT_ACROSS_GUARDIANS` — there's no split, there is only a single recovery key
- It isn't `K_OF_N_WITH_TIMELOCK` — no threshold, no timelock
- But it also isn't "unsupported" — the user can genuinely generate a self-custodied recovery key
- The meaningful self-sovereignty distinction is that **executing recovery requires a provider-hosted tool** (`keys.coinbase.com`) with no self-hosted alternative, so the user has a single point of failure on Coinbase's infrastructure availability. There is no other way to recover the wallet outside of this as they wouldn't have a seed phrase that would allow them to port this wallet to another provider specifically for the ERC 4337 implementation of the base app which is the latest version

Per the schema docstring's own guidance, this is a case where a new enum value is the right answer rather than forcing a misfit.

## Sketch of a proposed design

```ts
enum GuardianPolicyType {
    SECRET_SPLIT_ACROSS_GUARDIANS = '...',
    K_OF_N_WITH_TIMELOCK = '...',
    SELF_CUSTODY_WITH_PROVIDER_HOSTED_TOOL = '...', // NEW
}

enum RecoveryToolEnvironment {
    WEB,              // hosted webpage; depends on URL being live
    CLI,              // user-downloaded binary; runs offline
    NATIVE_APP,       // separate installed app; runs offline
    BROWSER_EXTENSION,
}

interface RecoveryTool {
    maintainedBy: Entity     // entity that authors/publishes/operates the tool
    environment: RecoveryToolEnvironment
    selfHostable: boolean    // can the user run their own copy if the maintainer disappears?
}

type GuardianPolicySelfCustodyWithProviderHostedTool = GuardianPolicyBase & {
    type: GuardianPolicyType.SELF_CUSTODY_WITH_PROVIDER_HOSTED_TOOL
    guardian: Guardian       // single self-custodied guardian
    recoveryTool: RecoveryTool
}
```

Note: The `selfHostable` boolean = `false` means the user has a single point of failure on the maintaining entity even with their key in hand; `true` means the recovery tool can be run locally without the entity's live infrastructure.

## Example usage

For Base App, this would look like:

```ts
accountRecovery: {
    guardianRecovery: supported({
        ref: 'https://help.coinbase.com/en/wallet/getting-started/smart-wallet-recovery',
        minimumGuardianPolicy: {
            type: GuardianPolicyType.SELF_CUSTODY_WITH_PROVIDER_HOSTED_TOOL,
            descriptionMarkdown:
                'The user generates and stores a recovery key (a standard Ethereum private key) which is registered on-chain as an additional owner of the Smart Wallet. Coinbase does not hold or back up the key. However, executing recovery requires accessing keys.coinbase.com — there is no self-hosted equivalent, so the recovery flow depends on Coinbase maintaining that service.',
            guardian: {
                type: GuardianType.SELF_CUSTODY,
            },
            recoveryTool: {
                maintainedBy: coinbase,
                environment: RecoveryToolEnvironment.WEB,
                selfHostable: false,
            },
        },
    }),
},
```

## Open questions for maintainer input

1. **Generalization scope:** Is this the right level? Are there adjacent recovery patterns the design should also accommodate (e.g. multi-key self-custody with multiple required tools, or tools with partial dependencies on a provider)?

2. **`RecoveryTool` as a top-level type:** Currently scoped to the new policy. Should it be reusable across other policy types in case future policies also have a tool-execution dependency?

3. **Rating heuristic** for the new type. The `selfHostable` boolean gives a clean two-tier rating:
   - `selfHostable: true` → user has true sovereignty → **PASS**
   - `selfHostable: false` → user has a single point of failure on the maintaining entity → **FAIL**

   For Base App specifically, this would land at **FAIL** for the `account-unruggability` rating.

The PR would also implement the `DATA_LOSS` and `ENTITY_TURNS_EVIL` scenario logic for this new type (lose the recovery key → unrecoverable; maintaining entity goes away → unrecoverable when `selfHostable: false`) rather than stubbing them, since Base App would actively use this type and unstubbed evaluators are needed to avoid crashing the rating system.
</div>

**Comment by polymutex** [comment]
There is similar logic in `src/schema/features/account-support.ts` with this enum meant to categorize tooling for smart account key rotation transaction creation capabilities:

```typescript
/** The ability (or lack thereof) to generate a transaction of a specific type. */
export enum TransactionGenerationCapability {
	/** The process to generate such a transaction relies on an external API. */
	RELYING_ON_EXTERNAL_API = 'RELYING_ON_EXTERNAL_API',

	/** The process to generate such a transaction requires the use of a standalone proprietary application. */
	USING_PROPRIETARY_STANDALONE_APP = 'USING_PROPRIETARY_STANDALONE_APP',

	/** The process to generate such a transaction requires the use of an open-source standalone application. */
	USING_OPEN_SOURCE_STANDALONE_APP = 'USING_OPEN_SOURCE_STANDALONE_APP',

	/** It is not possible to generate such a transaction. */
	IMPOSSIBLE = 'IMPOSSIBLE',
}
```

This is a very similar case. We could rename to enum to `AccountManagementTooling` and add `USING_PROPRIETARY_HOSTED_WEB_APP` and `USING_SELF_HOSTABLE_WEB_APP` to it, then use it in both places.

However, I'm coming around that this actually shouldn't be in the `GuardianPolicyType` enum. As the name implies, that is for guardian-based recovery. What Coinbase has here is **not** guardian-based recovery. It is just an alternate seed phrase that can rotate the keys of the account.

I think we should instead add a field in the `AccountRecovery` type (`src/schema/features/security/account-recovery.ts`) next to `guardianRecovery`, something like `alternateRecovery` which has fields like:

```typescript
accountRecovery: {
    guardianRecovery: notSupported, // There is no Guardian-based recovery feature in the Base App
    alternateRecovery: supported({
        ref: {
            url: 'https://help.coinbase.com/en/wallet/getting-started/smart-wallet-recovery',
            explanation: 'The user generates and stores a recovery key (a standard Ethereum private key) which is registered on-chain as an additional owner of the Smart Wallet. Coinbase does not hold or back up the key. However, executing recovery requires accessing keys.coinbase.com — there is no self-hosted equivalent, so the recovery flow depends on Coinbase maintaining that service.',
        },
        type: AccountManagementTooling.USING_PROPRIETARY_HOSTED_WEB_APP,
        entity: coinbase,
    }),
},
```

Rating logic would be pretty brutal though. The "Account recovery" attribute would be a `FAIL`, as there's no real recovery (device loss = all self-custody secrets lost = no recovery), and the "Account unruggability" attribute would also be a `FAIL`, as there's no self-sovereignty for the account recovery process. From a rating outcome perspective, this is basically the same outcome as just setting `guardianRecovery: notSupported` today and calling it a day.
The only effective difference is the explanation for why the rating would be produced.
